sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -457,6 +466,7 @@ public final int appendNull() {
   }
 
   public final int appendNotNull() {
+    assert (!(dataType() instanceof StructType)); // Use appendStruct()
```

Comment:
Previously, we have no user to use this API for `StructType`, right? I'm wondering if this is a breaking change of this API users.

Reply:
As mentioned in the PR description, I think it's fine since `WritableColumnVector` is a Spark internal API. Also see some [previous discussion](https://github.com/apache/spark/pull/34659#discussion_r769131591).